homeassistant/components/thermostat/homematic.py

```diff
@@ -29,20 +36,29 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     """Setup the Homematic thermostat."""
     devices = []
     try:
-        homegear = ServerProxy(config[CONF_ADDRESS])
         for name, device_cfg in config[CONF_DEVICES].items():
+            address = config[CONF_ADDRESS]
+            homegear = ServerProxy(address)
             # get device description to detect the type
             device_type = homegear.getDeviceDescription(
                 device_cfg[CONF_ID] + ':-1')['TYPE']
 
             if device_type in ['HM-CC-RT-DN', 'HM-CC-RT-DN-BoM']:
-                devices.append(HomematicThermostat(homegear,
+                devices.append(HomematicThermostat(TYPE_HM_THERMOSTAT,
```

Comment:
The only difference here between all `if` statements is the type. I would suggest you create a dictionary mapping device description to types:
```python
TYPES = {
    'HM-CC-RT-DN': TYPE_HM_THERMOSTAT,
    …
}

if device_type not in TYPES:
    raise ValueError('Device type not supported')
```